LAPACKE/src/lapacke_cgedmd_work.c

```diff
@@ -0,0 +1,175 @@
+                                lapack_complex_float* res, lapack_complex_float* b,
+                                lapack_int ldb, lapack_complex_float* w,
+                                lapack_int ldw, lapack_complex_float* s, lapack_int lds,
+                                lapack_complex_float* work, lapack_int lwork,
+                                lapack_complex_float* iwork, lapack_int liwork )
+{
+    lapack_int info = 0;
+    if( matrix_layout == LAPACK_COL_MAJOR ) {
+        /* Call LAPACK function and adjust info */
+        LAPACK_cgedmd( &jobs, &jobz, &jobf, &whtsvd, &m, &n, x, &ldx, y, &ldy,
+                       &k, reig, imeig, z, &ldz, res, b, &ldb, w, &ldw, s, &lds,
+                       work, &lwork, iwork, &liwork, &info );
+        if( info < 0 ) {
+            info = info - 1;
+        }
+    } else if( matrix_layout == LAPACK_ROW_MAJOR ) {
+        lapack_int ldx_t = MAX(1,m);
+        lapack_int ldy_t = MAX(1,m);
+        lapack_int ldz_t = MAX(1,m);
+        lapack_int ldb_t = MAX(1,m);
+        lapack_int ldw_t = MAX(1,m);
+        lapack_int lds_t = MAX(1,m);
+        lapack_complex_float* x_t = NULL;
+        lapack_complex_float* y_t = NULL;
+        lapack_complex_float* z_t = NULL;
+        lapack_complex_float* b_t = NULL;
+        lapack_complex_float* w_t = NULL;
+        lapack_complex_float* s_t = NULL;
+        /* Check leading dimension(s) */
+        if( ldx < n ) {
+            info = -5;
+            LAPACKE_xerbla( "LAPACKE_cgedmd_work", info );
+            return info;
+        }
+        if( ldy < n ) {
+            info = -5;
+            LAPACKE_xerbla( "LAPACKE_cgedmd_work", info );
+            return info;
+        }
+        if( ldz < n ) {
+            info = -5;
+            LAPACKE_xerbla( "LAPACKE_cgedmd_work", info );
+            return info;
+        }
+        if( ldb < n ) {
+            info = -5;
+            LAPACKE_xerbla( "LAPACKE_cgedmd_work", info );
+            return info;
+        }
+        if( ldw < n ) {
+            info = -5;
+            LAPACKE_xerbla( "LAPACKE_cgedmd_work", info );
+            return info;
+        }
+        if( lds < n ) {
+            info = -5;
+            LAPACKE_xerbla( "LAPACKE_cgedmd_work", info );
+            return info;
+        }
+        /* Query optimal working array(s) size if requested */
+        if( lwork == -1 ) {
+            LAPACK_cgedmd( &jobs, &jobz, &jobf, &whtsvd, &m, &n, x, &ldx, y, &ldy,
+                           &k, reig, imeig, z, &ldz, res, b, &ldb, w, &ldw, s, &lds,
+                           work, &lwork, iwork, &liwork, &info );
+            return (info < 0) ? (info - 1) : info;
+        }
+        /* Allocate memory for temporary array(s) */
+        x_t = (lapack_complex_float*)LAPACKE_malloc( sizeof(lapack_complex_float) * ldx_t * MAX(1,n) );
+        y_t = (lapack_complex_float*)LAPACKE_malloc( sizeof(lapack_complex_float) * ldy_t * MAX(1,n) );
```

Comment:
As far as I know, LAPACKE should interleave the calls to malloc with the error checks to avoid memory leaks if some, but not all of the mallocs succeed. 
```C
  /* Allocate memory for temporary array(s) */
  x_t = (lapack_complex_float*)LAPACKE_malloc( sizeof(lapack_complex_float) * ldx_t * MAX(1,n) );
  if( x_t == NULL ) {
      info = LAPACK_TRANSPOSE_MEMORY_ERROR;
      goto exit_level_0;
  }
  y_t = (lapack_complex_float*)LAPACKE_malloc( sizeof(lapack_complex_float) * ldy_t * MAX(1,n) );
  if( y_t == NULL ) {
     info = LAPACK_TRANSPOSE_MEMORY_ERROR;
     goto exit_level_1;
  }
  [...]
  /* Release memory and exit */
  [...]
exit_level_1:
  LAPACKE_free( x_t );
exit_level_0:
  if( info == LAPACK_TRANSPOSE_MEMORY_ERROR ) {
    LAPACKE_xerbla( "LAPACKE_cgedmd_work", info );
  }
```

Reply:
I see, thank you for pointing this out. I will update the wrappers accordingly.

Reply:
@angsch if you have the time can you confirm the changes pushed follow the example provided? I templated off of `LAPACKE_DGEQRF_WORK` and `exit_level_0:` was the only present. I just want to be certain it is ok to manage the `LAPACKE_free( );` in this way.

Reply:
The idea is to have a different jump label after each malloc. My my earlier comment was ambiguous. I am sorry about this. An example with multiple workspaces is `lapacke_dsysvx_work.c`.

With that applied, `LAPACKE/src/lapacke_cgedmdq_work.c` would read:
```C
       /* Allocate memory for temporary array(s) */
        f_t = (lapack_complex_float*)LAPACKE_malloc( sizeof(lapack_complex_float) * ldf_t * MAX(1,n) );
        if( f_t == NULL ) {
            info = LAPACK_TRANSPOSE_MEMORY_ERROR;
            goto exit_level_0;
        }
        x_t = (lapack_complex_float*)LAPACKE_malloc( sizeof(lapack_complex_float) * ldx_t * MAX(1,n) );
        if( x_t == NULL ) {
            info = LAPACK_TRANSPOSE_MEMORY_ERROR;
            goto exit_level_1;  // level_0 --> level_1
        }
        y_t = (lapack_complex_float*)LAPACKE_malloc( sizeof(lapack_complex_float) * ldy_t * MAX(1,n) );
        if( y_t == NULL ) {
            info = LAPACK_TRANSPOSE_MEMORY_ERROR;
            goto exit_level_2; // level_0 --> level_1
        }
        z_t = (lapack_complex_float*)LAPACKE_malloc( sizeof(lapack_complex_float) * ldz_t * MAX(1,n) );
        if( z_t == NULL ) {
            info = LAPACK_TRANSPOSE_MEMORY_ERROR;
            goto exit_level_3; // level_0 --> level_3
        }
        b_t = (lapack_complex_float*)LAPACKE_malloc( sizeof(lapack_complex_float) * ldb_t * MAX(1,n) );
        if( b_t == NULL ) {
            info = LAPACK_TRANSPOSE_MEMORY_ERROR;
            goto exit_level_4; // level_0 --> level_4
        }
        v_t = (lapack_complex_float*)LAPACKE_malloc( sizeof(lapack_complex_float) * ldv_t * MAX(1,n) );
        if( v_t == NULL ) {
            info = LAPACK_TRANSPOSE_MEMORY_ERROR;
            goto exit_level_5; // level_0 --> level_5
        }
        s_t = (lapack_complex_float*)LAPACKE_malloc( sizeof(lapack_complex_float) * lds_t * MAX(1,n) );
        if( s_t == NULL ) {
            info = LAPACK_TRANSPOSE_MEMORY_ERROR;
            goto exit_level_6; // level_0 --> level_6
        }
        /* Transpose input matrices */
        LAPACKE_cge_trans( matrix_layout, m, n, f, ldf, f_t, ldf_t );
        [...]
        LAPACKE_cge_trans( matrix_layout, m, n, s, lds, s_t, lds_t );
        /* Call LAPACK function and adjust info */
        LAPACK_cgedmdq( &jobs, &jobz, &jobr, &jobq, &jobt, &jobf, &whtsvd, &m,
                        &n, f, &ldf, x, &ldx, y, &ldy, &nrnk, tol, &k, reig,
                        imeig, z, &ldz, res, b, &ldb, v, &ldv, s, &lds,
                        work, &lwork, iwork, &liwork, &info );
        if( info < 0 ) {
            info = info - 1;
        }
        /* Transpose output matrices */
        LAPACKE_cge_trans( LAPACK_COL_MAJOR, m, n, f_t, ldf_t, f, ldf );
        [...]
        LAPACKE_cge_trans( LAPACK_COL_MAJOR, m, n, s_t, lds_t, s, lds );
        /* Release memory and exit */
exit_level_6:
exit_level_6:
exit_level_6:
exit_level_1:
        LAPACKE_free( f_t );
        LAPACKE_free( x_t );
        LAPACKE_free( y_t );
        LAPACKE_free( z_t );
        LAPACKE_free( b_t );
        LAPACKE_free( v_t );
        LAPACKE_free( s_t );
exit_level_0:
        if( info == LAPACK_TRANSPOSE_MEMORY_ERROR ) {
            LAPACKE_xerbla( "LAPACKE_cgedmdq_work", info );
        }
    } else {
        info = -1;
        LAPACKE_xerbla( "LAPACKE_cgedmdq_work", info );
    }
```

Reply:
Same as for the other comment in `lapacke_cgedmd_work`, I believe this is resolved.

I have propagated these changes to all precisions as well. Thank you again for your help here!